pkg/tsdb/loki/api.go

```diff
@@ -25,10 +25,27 @@ func newLokiAPI(client *http.Client, url string, log log.Logger) *LokiAPI {
 	return &LokiAPI{client: client, url: url, log: log}
 }
 
+func toLokiDirection(direction Direction) (string, error) {
+	switch direction {
+	case "backward":
+		return "BACKWARD", nil
+	case "forward":
+		return "FORWARD", nil
+	default:
+		return "", fmt.Errorf("invalid direction: %v", direction)
+	}
+}
+
 func makeRequest(ctx context.Context, lokiDsUrl string, query lokiQuery) (*http.Request, error) {
 	qs := url.Values{}
 	qs.Set("query", query.Expr)
 
+	direction, err := toLokiDirection(query.Direction)
```

Comment:
As `query.Direction` is type of `Direction` which is type of `string`, can't we just use `string(query.Direction)` to set direction?

Reply:
the `Direction` type uses lowercase-strings (because that's how the other constants are too in https://github.com/grafana/grafana/blob/main/pkg/tsdb/loki/types.go ,

and the Loki HTTP API says `direction` should be uppercase, so i have to switch things.

there are a couple alternatives of course:
- just use https://pkg.go.dev/strings#ToUpper to uppercase-convert the string
- just send lowercase strings to Loki, maybe it is not case sensitive

i chose the current approach because:
- it felt the least "smart" 😄 
- i like that it's done in a "defensive" way. if, for some reason the caller sends an invalid string in `Direction`, it will return a good error message

but of course, it means more lines of code and i sometimes overcomplicate things, so i'm open to suggestions :)

Reply:
>and the Loki HTTP API says direction should be uppercase, so i have to switch things.

Where did you find this? Because I checked https://grafana.com/docs/loki/latest/api/#get-lokiapiv1query_range and it says that the `Supported values are forward or backward. Defaults to backward.` I also tested this with lower case and it worked.

Reply:
sorry, my mistake :) you're right, it is lowercase. i'll adjust the code.